components/script/dom/htmlcollection.rs

```diff
@@ -282,7 +282,14 @@ impl HTMLCollection {
         after
             .following_nodes(&self.root)
             .filter_map(DomRoot::downcast)
-            .filter(move |element| self.filter.filter(&element, &self.root))
+            .filter(move |element: &DomRoot<Element>| {
+                if element.upcast::<Node>().is_in_shadow_tree() &&
+                    element.upcast::<Node>().owner_shadow_root() != self.root.owner_shadow_root()
+                {
+                    return false;
```

Comment:
How can this happen? I don't think it can.

Reply:
Nested shadow trees, but I guess we can ignore that for now.

Reply:
I don't think following_nodes digs into shadow trees, right?